mono/mini/mini-exceptions.c

```diff
@@ -2881,19 +2881,13 @@ mono_print_thread_dump_internal (void *sigctx, MonoContext *start_ctx)
 	MonoContext ctx;
 #endif
 	GString* text;
-	char *name;
-	GError *error = NULL;
 
 	if (!thread)
 		return;
 
 	text = g_string_new (0);
-	if (thread->name) {
-		name = g_utf16_to_utf8 (thread->name, thread->name_len, NULL, NULL, &error);
-		g_assert (!error);
-		g_string_append_printf (text, "\n\"%s\"", name);
-		g_free (name);
-	}
+	if ((int)thread->name_len > 0) // A little wonky with large numbers but ok.
```

Comment:
Why the `(int)` cast?

Reply:
Printf takes int here -- a mistake, should have been a size_t. We have a uint32. This is actually an argument in favor of nul termination-- but what about embedded nuls? Embedded nuls are another reason to pass lengths and not use strlen. Printf will still be confused though.